### ISSUE TITLE
"Notebook: Format Notebook" with Black completely mangles the notebook

### ISSUE THREAD

**Issue by ldorigo** [microsoft/vscode-jupyter#7812]
Issue Type: <b>Bug</b>

Hello, 

Formatting single cells with black works fine. I tried using the "format notebook" command, and it completely mangled my notebook - replacing random lines left and right and causing over an hour of work to recover my work (which is partly my fault - just mentioning it to point out it's a relatively bad bug). 

I can provide the notebook on which this happens, it contains slightly private data so don't want to attach it publicly on this issue. 

I've found an old issue https://github.com/microsoft/vscode-jupyter/issues/1060, which stated that formatting entire notebooks wasn't implemented yet - don't know if it's still valid? If that is the case, the command should be disabled until the  functionality is implemented - rather than risking lots of lost work.

Extension version: 2021.8.2041215044
VS Code version: Code 1.60.2 (7f6ab5485bbc008386c4386d08766667e155244e, 2021-09-22T12:01:43.795Z)
OS version: Linux x64 5.14.6-zen1-1-zen
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 4800H with Radeon Graphics (16 x 1400)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 2|
|Memory (System)|30.78GB (2.86GB free)|
|Process Argv|--crash-reporter-id 9614dc43-c276-4d0e-94fc-c84de505d77f|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma-i3|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE-i3|
|XDG_SESSION_TYPE|x11|
</details>

**Comment by IanMatthewHuff** [comment]
@ldorigo I don't believe that you actually need to provide your file here. I just played with this a bit myself, and while it worked on some very basic files I was able to get some very mangled results.

**Comment by greazer** [comment]
This is a dupe of: https://github.com/microsoft/vscode-python/issues/16980. Should be fixed in the latest Python insiders extension. Will be released in a point release.
